### ISSUE TITLE
Typescript Errors In New Project (5.7.1)

### ISSUE THREAD

**Issue by HockeyJustin** [polkadot-js/api#3919]
Hi,

I'm trying to learn Polkadot. 

When I install polkadot.js via npm and do `ng serve` (just run the project), I get a ton of typescript errors that prevent the project from building.

e.g

> ERROR in node_modules/@polkadot/types/types/detect.d.ts:9:62 - error TS1110: Type expected.
> 
> 9 export declare type __Sanitize<K extends string> = K extends ` ${infer X}` | `${infer X} ` | ` ${infer X} ` ? __Sanitize<X> : K extends `${infer A} ${infer B}` ? __Sanitize<`${A}${B}`> : K extends `${infer A};${string}]${infer B}` ? __Sanitize<`${A}${B}`> : K extends `${infer X};${string}]` ? __Sanitize<X> : K;

I've tried changing my tsconfig to ignore node_module errors like this, but no joy

```
{
  "compileOnSave": false,
  "compilerOptions": {
    "baseUrl": "./",
    "outDir": "./dist/out-tsc",
    "sourceMap": true,
    "declaration": false,
    "downlevelIteration": true,
    "experimentalDecorators": true,
    "module": "esnext",
    "moduleResolution": "node",
    "importHelpers": true,
    "target": "es2015",
    "skipLibCheck": true,
    "lib": [
      "es2018",
      "dom"
    ]
  },
  "exclude":[
    "**/node_modules/"
  ],
  "angularCompilerOptions": {
    "fullTemplateTypeCheck": true,
    "strictInjectionParameters": true
  }
}
```

Here's my [test project](https://github.com/HockeyJustin/PolkadotLearning), which can be run by a `git clone`, then `npm install` and then `ng serve`.

Is this an error in the current release, or am I doing something wrong?

**Comment by jacogr** [comment]
This is your issue - https://github.com/HockeyJustin/PolkadotLearning/blob/main/PolkadotLearning/package.json#L46

The above errors are due to template strings, which has been introduced in the TS 4 series. (Additionally the rxjs dependency polkadot-js uses needs at least TS 4.2)

**Comment by HockeyJustin** [comment]
@jacogr  - Thanks for the quick response. I'll give that a go after work and add details here for future searchers.

**Comment by HockeyJustin** [comment]
In the end, I couldn't use Typescript 4.2. I needed 4.4.2 to get it going.

Unfortunately Angular 12/13.next had issues with that. 

I tried using Vue.js and it all works nicely. Vue also handles the async/await much better, so I'm happy to move over.

Thanks for the pointer. I was going to suggest me creating a PR for updating the docs, but it'd soon be out of date, so hopefully this issue will steer someone in the right direction.

Thanks for your help.

I had to
